ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -1228,21 +1225,28 @@ func compareRayClusterJsonHash(spec1 rayv1.RayClusterSpec, spec2 rayv1.RayCluste
 
 // isHeadPodRunningAndReady checks if the head pod of the RayCluster is running and ready.
 func (r *RayServiceReconciler) isHeadPodRunningAndReady(ctx context.Context, instance *rayv1.RayCluster) (bool, error) {
+	headPod, err := r.getHeadPod(ctx, instance)
+	if err != nil {
+		return false, err
+	}
+	return utils.IsRunningAndReady(headPod), nil
+}
+
+func isServeAppUnhealthyOrDeployedFailed(appStatus string) bool {
+	return appStatus == rayv1.ApplicationStatusEnum.UNHEALTHY || appStatus == rayv1.ApplicationStatusEnum.DEPLOY_FAILED
+}
+
+func (r *RayServiceReconciler) getHeadPod(ctx context.Context, instance *rayv1.RayCluster) (*corev1.Pod, error) {
```

Comment:
In the KubeRay codebase, several places need to retrieve the head Pod. We may move this function to `util.go`, and then always use this function to retrieve the head Pod in this PR or a follow up PR.